R/dv8.R

```diff
@@ -0,0 +1,254 @@
+#' Creates a gitlog numstat file from a git file
+#'
+#' From a git file *.git, a gitlog numstat file *.txt is created
+#'
+#' @param git_repo_path path to local repository
+#' @param git_numstat_path file name of git log output
+#'
+#' @return path to git log output
+#' @export
+#' @seealso \code{\link{dv8_gitnumstat_to_hdsmb}} for creating hsdsmb binary file from a git numstat text file
+#'
+dv8_gitlog_to_gitnumstat <- function(git_repo_path, git_numstat_path) {
+  system2(command = "git",
+          args = c("--git-dir",
+                   git_repo_path,
+                   "log",
+                   "--numstat",
+                   "--date=iso"),
+          stdout = git_numstat_path
+          )
+  return(git_numstat_path)
+}
+
+
+#' Converts a git log text file into a hdsmb binary file
+#'
+#' An input file *.txt is converted to *.dv8-dsm
+#'
+#' @param dv8_path path to DV8 binary
+#' @param git_numstat_path path to local git log text file obtained by \code{\link{dv8_gitlog_to_gitnumstat}}
+#' @param start start of date/time range in ISO format ie. 2017-07-08T00:00:00Z
+#' @param stop end of date/time range in ISO format ie. 2018-07-08T00:00:00Z
+#' @param max_cochange_count maximum count of co-changed files per commit, default 1000
+#' @param hdsmb_path name of output hdsmb binary file, defaults to name of input
+#' @param params_output_file output file used to record parameters
+#'
+#' @return path to hdsmb binary file
+#' @export
+#' @seealso \code{\link{dv8_hsdsmb_to_decoupling_level}} to calculate decoupling level from hsdsmb binary file
+#' @seealso \code{\link{dv8_hsdsmb_to_hierclsxb}} to compute design rule hierarchy from hsdsmb binary file
+#' @seealso \code{\link{dv8_hsdsmb_drhier_to_excel}} to export a hsdsmb binary file as an excel file
+#' @seealso \code{\link{dv8_hsdsmb_dsmb_to_dsmj}} to export a hsdsmb binary file as a json file
+#'
+dv8_gitnumstat_to_hdsmb <- function(dv8_path,
+                                    git_numstat_path,
+                                    hdsmb_path,
+                                    start,
+                                    stop,
+                                    max_cochange_count=1000,
+                                    params_output_file="") {
+
+  dv8_path <- path.expand(dv8_path)
+  output_file_arg <- sprintf("-outputFile %s", hdsmb_path)
+  start_arg <- sprintf("-start %s", start)
+  stop_arg <- sprintf("-stop %s", stop)
+  max_cochange_count_arg <- sprintf("-maxCochangeCount %i", max_cochange_count)
+  params_output_file_arg <- ""
+
+  if (params_output_file != "") {
+    params_output_file_arg <- sprintf("-paramsOutputFile %s ", params_output_file)
+  }
+
+  system2(command=dv8_path,
+          args=c("scm:history:gittxt:convert-matrix",
+                 start_arg,
+                 stop_arg,
```

Comment:
These parameters are optional. You have to include them in the function so the user may use, but you must also include default values so it works if they are not used.